### PR TITLE
updated neo4j details

### PR DESCRIPTION
**Pull request by akollegger**
Updated to reflect Neo4j's capabilities. There are a few items that could be discussed:

facets: faceted search is possible with Neo4j but not as a feature of the vector index itself. I've marked this as "none"
embeddings_structured: we naturally support structured data but not yet as a vector. Marked as "full"
Happy to discuss these or others aspects, intending to honor the intention of the fields.

Best,
Andreas

### CODE REVIEW

File: docs/tools/vdb_table/data/neo4j.json [hunk 1]
```diff
@@ -48,7 +49,7 @@
     "comment": ""
   },
   "multi_vec": {
-    "support": "none",
+    "support": "full",
     "source_url": "",
```
Comment (prrao87) [inline docs/tools/vdb_table/data/neo4j.json:53]: I think users would want to see what kind of hybrid search is available in Neo4j. I wasn't able to find a source URL that describes this in the Neo4j documentation page, but I did see an example of how to use this in the LangChain [docs](https://python.langchain.com/docs/integrations/vectorstores/neo4jvector#hybrid-search-vector-keyword). Is there a source URL (or maybe even link to code on GitHub) that's from Neo4j that guides the user on learning more about hybrid search in Neo4?

File: docs/tools/vdb_table/data/neo4j.json [hunk 2]
```diff
@@ -123,30 +124,30 @@
     "comment": ""
   },
   "disk_index": {
-    "support": "none",
+    "support": "full",
     "source_url": "",
     "comment": ""
```
Comment (prrao87) [inline docs/tools/vdb_table/data/neo4j.json:129]: A description or source URL showing that the index is held on disk (without being in memory) would be required here. To my knowledge, Neo4j's vector index is held in memory (even though the vectors themselves are on disk).  Other vendors that have this option checked are pointing to docs that show usage of DiskANN, SAI indexes, and so on, so this should be marked "none" imo, unless there's a documentation page (or code) that you could link to showing the on-disk index with a description.

**Review by prrao87 — CHANGES_REQUESTED** [review]
Small changes/updates/clarifications needed, thanks!

**Comment by prrao87** [comment]
> facets: faceted search is possible with Neo4j but not as a feature of the vector index itself. I've marked this as "none"
> embeddings_structured: we naturally support structured data but not yet as a vector. Marked as "full"

Both these points make sense, fully agree.

**Comment by svonava** [comment]
@prrao87 structured_embeddings is about actually calculating the embeddings of the structured data - so I don't think this is full/partial?

**Comment by svonava** [comment]
@akollegger for the structured embeddings, I now double checked and it seems that neo4j supports the node embeddings which is great! What about the rest of the node metadata beyond the graph structure? Like if I store eg categorical data or numeric parameters on the node, could I vectorize that right along the semantic/graph embeddings?

That would qualify as full support - if you only do the graph side of it, I'd call it partial. But curious also what @dhruv-anand-aintech thinks!

**Comment by prrao87** [comment]
That's well explained, @svonava-superlinked. I think I agree with the "partial" if there's no way to vectorize structured representations of metadata (not just just one single metadata field's unstructured blobs)

File: docs/tools/vdb_table/data/neo4j.json [hunk 3]
```diff
@@ -14,26 +14,27 @@
...
+  "github_stars": 12200,
+  "vector_launch_year": 2023,
   "metadata_filter": {
     "support": "full",
     "source_url": "",
     "comment": ""
   },
   "hybrid_search": {
-    "support": "none",
+    "support": "full",
```
Comment (dhruv-anand-aintech) [inline docs/tools/vdb_table/data/neo4j.json:37]: please provide a reference to docs/example notebook for this in "source_url". It should show ability to merge between ranked lists of keyword search and vector search results.

File: docs/tools/vdb_table/data/neo4j.json [hunk 4]
```diff
@@ -123,30 +124,30 @@
     "comment": ""
   },
   "disk_index": {
-    "support": "none",
+    "support": "full",
```
Comment (dhruv-anand-aintech) [inline docs/tools/vdb_table/data/neo4j.json:127]: please provide a reference to docs/example notebook for this in "source_url" as requested by @prrao87 below

File: docs/tools/vdb_table/data/neo4j.json [hunk 5]
```diff
@@ -14,26 +14,27 @@
...
-      ""
+      "Java",
+      "Scala"
     ],
     "source_url": "",
     "comment": ""
   },
-  "github_stars": 0,
-  "vector_launch_year": 0,
+  "github_stars": 12200,
```
Comment (dhruv-anand-aintech) [inline docs/tools/vdb_table/data/neo4j.json:29]: The github stars are for open-source repo for the vector search engine/db itself. Please reset this. Thanks.

**Review by dhruv-anand-aintech — CHANGES_REQUESTED** [review]
please address comments and request review

File: docs/tools/vdb_table/data/neo4j.json (hunk 5 above)
Comment (prrao87) [reply docs/tools/vdb_table/data/neo4j.json:29]: Shouldn't there also be a `source_url` here to show what repo we're referring to?

**Comment by svonava** [comment]
@akollegger @dhruv-anand-aintech would love to get this merged - for the items that don't have a source, let's revert to original value for now and merge the rest, what do you think?

File: docs/tools/vdb_table/data/neo4j.json [hunk 6]
```diff
@@ -78,13 +79,13 @@
     "comment": ""
   },
   "embeddings_structured": {
-    "support": "",
+    "support": "full",
     "source_url": "",
-    "comment": ""
+    "comment": "Graphs are kinda our thing :)"
```
[the remark this answers is not in the document]
Comment (dhruv-anand-aintech) [reply docs/tools/vdb_table/data/neo4j.json:84]: It is a graph database, but for this there'd have to be an example of where the node/edge embeddings are being calculated in the DB

**Comment by dhruv-anand-aintech** [comment]
Added source where it was clear, reverted the rest which didn't have source. Merging now.

@akollegger please create a new PR with the source urls included. Thanks.

**Comment by svonava** [comment]
Actually for the structured data embeddings, neo4j should have partial, because the graph embeddings (node2vec and graphsage) are 1st-class citizens:
Overview:
https://neo4j.com/developer/graph-data-science/graph-embeddings/
Example built in node embedding algo:
https://neo4j.com/docs/graph-data-science/current/machine-learning/node-embeddings/node2vec/

**Pull request merged**